### ISSUE TITLE
replace vgg with resnet

### ISSUE THREAD

**Issue by lyxbyr** [skhadem/3D-BoundingBox#14]
Hi 
Thanks for your great work, I am now trying to replace the backbone of second stage using Resnet since Resnet usually has a better performance when doing a same work. However, after I replace the vgg by resnet, the result is terrible. I wonder if you have do the same thing and cloud you please tell me whether this idea is useful. Thank you!
 
Best regards

**Comment by Cknex** [comment]
Hi, I also tried Resnet18 and Resnet50, but got a similar performance to VGG.

**Comment by lyxbyr** [comment]
@Cknex Hi ， similar performance how to eval it ? computer dimension error and orientation error ？the final three branch fc layer node need to change,eg resnet18 or resnet50  the three branch fc you set many fc node.

**Comment by lyxbyr** [comment]
![Screenshot from 2020-03-28 11-55-54](https://user-images.githubusercontent.com/55404168/77814393-9bb24b00-70eb-11ea-82e0-80ac73828acb.png)

**Comment by lyxbyr** [comment]
my result:
               Angle error  Dimension error
Vgg19     13.683045   0.283335
resnet34  27.887893   0.849132
resnet50  21.706590   0.821558

**Comment by Cknex** [comment]
I just use the mean absolute error,
For resnet50, the angle error is 0.77, and the dimension error is [0.13834581  0.10448476  0.55814862]. For angle, the FC layers are 2048*7*7->256,  256->256, 256->256*2. For dimension, the FC layers are 2048*7*7->512, 512->512, 512->3   
**However, the model doesn't learn anything. It almost always output the mean value....**

**Comment by Cknex** [comment]
Sorry, I made a mistake. My performance of Resnet50 is also bad.

I use the mean absolute error, and only test for 'Car'
                     Angle error,  Dimension error
VGG19               0.108         [0.086 0.067 0.194]
Resnet50            0.534         [0.100 0.076  0.320]

As for the FC layers, I just simply adapt the first fc layer to be compatible with the output of Resnet50

**Comment by lyxbyr** [comment]
Hi,for resnet50,fc like this result may be better  : self.orientation = nn.Sequential(
            nn.Linear(2048 * 7 * 7, 1024),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(1024, 1024),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(1024, bins * 2)  # to get sin and cos
        )
        self.confidence = nn.Sequential(
            nn.Linear(2048 * 7 * 7, 1024),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(1024, 1024),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(1024, bins),
            # nn.Softmax()
            # nn.Sigmoid()
        )
        self.dimension = nn.Sequential(
            nn.Linear(2048 * 7 * 7, 512),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(512, 512),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(512, 3)
        )

**Comment by Cknex** [comment]
Thanks very much!

**Comment by CaoBo1996** [comment]
@Cknex hey，i found your result 
Angle error, Dimension error
VGG19 0.108 [0.086 0.067 0.194]
Resnet50 0.534 [0.100 0.076 0.320]

Angle error was very good,but in my test,my result is bad.
i split kitti train set into split_train and split_val
i train in split_train，and test in split_val 
but my result is bad 
i also use mean absolute error,and test in Car
this is my result 
Car angle error：0.4860737579686576
so can you help me why?

**Comment by Cknex** [comment]
@CaoBo1996 
Sorry that I have lost my code. In memory I didn't change much. Maybe just more units in middle layers.
